File: javaCode/src/com/acme/MaxSubArray.java
package com.acme;

public class MaxSubArray {
    public int maxSubArray(int[] nums) {
        int prev = Integer.MIN_VALUE;
        int sum = 0;

        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            prev = sum > prev ? sum : prev;
            sum = sum > 0 ? sum : 0;
        }

        return prev;
    }
}

File: cpp/MaxSubarray.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int len = nums.size();
        int dp[len];
        dp[0] = nums[0];
        int max = dp[0];
        for (int i = 1; i<len; i++) {
            dp[i] = nums[i] + (dp[i-1]>0? dp[i-1]:0);
            max = max>dp[i]? max:dp[i];
        }
        return max;
    }
};